api/modules/budgeting.js

```diff
@@ -0,0 +1,103 @@
+const db = require('../models');
+const moment = require('moment')
+
+const budgeting = module.exports = (() => {
+
+    const createPayment = async ({ paymentInformation }) => {
+        const client = await getClientWithExternalId({ id: paymentInformation.customer_id })
+        if (client) {
+            return db.models.Payment.create({
+                amount: paymentInformation.amount,
+                external_uuid: paymentInformation.external_uuid,
+                date_incurred: moment(paymentInformation.date_incurred['_d']).format('YYYY-MM-DD HH:mm:ss'),
+                date_paid: paymentInformation.date_paid ? moment(paymentInformation.date_paid['_d']) : null,
+                client_id: client.id,
+                external_uuid_type: paymentInformation.external_uuid_type
+            })
```

Comment:
Suggestion: deconstruct above to reduce repeated references to `paymentInformation`

```
const {
    amount,
    external_uuid,
    date_incurred,
    date_paid,
    amount,
} = paymentInformation
    
return db.models.Payment.create({
    amount: amount,
    external_uuid: external_uuid,
    date_incurred: moment(date_incurred['_d']).format('YYYY-MM-DD HH:mm:ss'),
    date_paid: date_paid ? moment(date_paid['_d']) : null,
    client_id: client.id,
    external_uuid_type: external_uuid_type
})
```

which of course then simplifies to

```
return db.models.Payment.create({
    amount,
    external_uuid,
    date_incurred: moment(date_incurred['_d']).format('YYYY-MM-DD HH:mm:ss'),
    date_paid: date_paid ? moment(date_paid['_d']) : null,
    client_id: client.id,
    external_uuid_type
})
```